### PR TITLE
Add Reset, clock & language pids to the moving light.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/moving_light.c [hunk 1]
```diff
@@ -55,15 +61,109 @@ typedef struct {
...
+      return 31;
+      break;
+    case 2:
+      return is_leap_year ? 29 :28;
+      break;
+    case 4:
+    case 6:
+    case 9:
+    case 11:
+      return 31;
```
Comment (peternewman) [inline firmware/src/moving_light.c]: 30!

File: firmware/src/moving_light.c [hunk 2]
```diff
@@ -55,15 +61,109 @@ typedef struct {
...
+      break;
+    default:
+      return 0;
+  }
+}
+
 // PID Handlers
 // ----------------------------------------------------------------------------
+int MovingLightModel_GetLanguageCapabilties(const RDMHeader *header,
```
Comment (peternewman) [inline firmware/src/moving_light.c]: SPaG Capabilities

File: firmware/src/moving_light.c [hunk 3]
```diff
@@ -221,6 +321,65 @@ int MovingLightModel_SetPowerState(const RDMHeader *header,
...
+
+int MovingLightModel_SetClock(const RDMHeader *header,
+                              const uint8_t *param_data) {
+  if (header->param_data_length != sizeof(uint16_t) + 5 * sizeof(uint8_t)) {
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  const uint16_t year = ExtractUInt16(param_data);
+  const uint8_t month = param_data[2];
+  if (year < 2003u || month == 0u || month > 12u) {
```
Comment (peternewman) [inline firmware/src/moving_light.c]: We don't allow older than 2003?

File: firmware/src/moving_light.c [hunk 4]
```diff
@@ -279,6 +447,32 @@ static void MovingLightModel_Tasks() {
     g_moving_light.lamp_state = LAMP_ON;
     g_moving_light.lamp_strikes++;
   }
+
+  if (CoarseTimer_HasElapsed(g_moving_light.clock_timer, ONE_SECOND)) {
```
Comment (peternewman) [inline firmware/src/moving_light.c]: We can do burn in now you're tracking this.

File: firmware/src/moving_light.c [hunk 5]
```diff
@@ -30,15 +31,20 @@
...
 
 static const unsigned int LAMP_STRIKE_DELAY = 50000u;
+static const uint32_t ONE_SECOND = 10000;
 static const char DEVICE_MODEL_DESCRIPTION[] = "Ja Rule Moving Light";
 static const char SOFTWARE_LABEL[] = "Alpha";
 static const char DEFAULT_DEVICE_LABEL[] = "Default Label";
 static const char PERSONALITY_DESCRIPTION1[] = "8-bit mode";
 static const char PERSONALITY_DESCRIPTION2[] = "16-bit mode";
 
+static const char LANGUAGE_ENGLISH[] = "en";
```
Comment (peternewman) [inline firmware/src/moving_light.c]: Store these centrally.

Comment (nomis52) [inline firmware/src/moving_light.c]: Right now they are only used in this model. If we go beyond that I'll move them to a common file.

File: firmware/src/moving_light.c (hunk 1 above)
Comment (nomis52) [inline firmware/src/moving_light.c]: Done

File: firmware/src/moving_light.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/moving_light.c]: Done

File: firmware/src/moving_light.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/moving_light.c]: Nope. That's from directly from the standard.

File: firmware/src/moving_light.c (hunk 4 above)
Comment (nomis52) [inline firmware/src/moving_light.c]: That's in the dimmer model, not the moving light one.

File: firmware/src/moving_light.c (hunk 3 above)
Comment (peternewman) [inline firmware/src/moving_light.c]: So I see.

File: firmware/src/moving_light.c (hunk 4 above)
Comment (peternewman) [inline firmware/src/moving_light.c]: We could add burn in to this. :)

File: firmware/src/moving_light.c (hunk 5 above)
Comment (peternewman) [inline firmware/src/moving_light.c]: Okay

**Comment by peternewman** [comment]
LGTM

**Pull request merged**